### ISSUE TITLE
Trigger: QuickFix menu on a problem

### ISSUE THREAD

**Issue by cahaseler** [ai-genie/chatgpt-vscode#31]
### Describe the feature

Hi!

I wrote this over the weekend because taking the time to copy an error message is just too much work for me:

https://marketplace.visualstudio.com/items?itemName=haselerdev.aiquickfix

But - even GPT-4 isn't quite good enough to be consistently able to fix problems every time. I'd rather have the tool as a way to prompt Genie then decide if I like it or need to iterate. You interested in me doing a PR to add this?

**Comment by cahaseler** [comment]
Oh, I see that this github doesn't actually have the source code - but still, happy to help if you want it, or feel free to grab relevant code from my repo if you'd rahter incorporate it yourself.

**Comment by genieai-info** [comment]
@cahaseler this is interesting but correct we use this repo as a docs-only repository. Did you developed this extension to be a companion to Genie? Do you mind sharing a demo recording with us how it works?

**Comment by cahaseler** [comment]
Sure, will record a video a little later. I developed it to fill a gap and to be used alongside Genie, or just as a standalone simple tool without all the complex functionality.

All it does is add a button to the little lightbulb "QuickFix" menu that appears on "problems" identified by linters/typescript - when you click it, it grabs the error message and the block of code around the error, and passes them to GPT, then writes back the result over the initial problem. It mostly works, but there's also times when I wish it would just use that menu as a prompt for Genie instead, because even GPT-4 isn't perfect.

I developed it because I found I was copy-pasting error messages from my problem list into the "ad-hoc" prompt for Genie a lot. And I wanted to see how easy it was to develop a vscode extension with GPT's help as a weekend project. :)
